merkledag.go

```diff
@@ -10,16 +10,23 @@ import (
 	bserv "github.com/ipfs/go-blockservice"
 	cid "github.com/ipfs/go-cid"
 	ipldcbor "github.com/ipfs/go-ipld-cbor"
-	ipld "github.com/ipfs/go-ipld-format"
+	format "github.com/ipfs/go-ipld-format"
+	legacy "github.com/ipfs/go-ipld-legacy"
+	dagpb "github.com/ipld/go-codec-dagpb"
+	_ "github.com/ipld/go-ipld-prime/codec/raw"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
 )
 
 // TODO: We should move these registrations elsewhere. Really, most of the IPLD
 // functionality should go in a `go-ipld` repo but that will take a lot of work
 // and design.
 func init() {
-	ipld.Register(cid.DagProtobuf, DecodeProtobufBlock)
-	ipld.Register(cid.Raw, DecodeRawBlock)
-	ipld.Register(cid.DagCBOR, ipldcbor.DecodeBlock)
+	format.Register(cid.DagProtobuf, DecodeProtobufBlock)
+	format.Register(cid.Raw, DecodeRawBlock)
+	format.Register(cid.DagCBOR, ipldcbor.DecodeBlock)
```

Comment:
Why was this here before and do we need it now? Mostly noticing the asymmetry in that the other two codecs have legacy types and this one doesn't.

Reply:
I honestly have no idea why DagCBOR is being registered in a library that contains only a protobuf implementation. But I don't know the history.

But these are needed for go-ipld-format to continue working I think.

As for the legacy types, there is a default legacy type implementation in go-ipld-legacy that will work for CBOR and other formats.

The reason these special types were needed is that they implement MORE than the go-ipld-format spec and the logic of some of the go-ipld-format methods differs from the standard implementation (namely Resolve & ResolveLink which search the links for a matching name)

Reply:
Conclusion: We're leaving these in as convenience functions for people who have been relying on registration happening here.

TODO: Let's file an issue for removing these